File: backend/app/schemas/manga_local.py
"""
本地漫画库 Schema
"""
from datetime import datetime
from typing import Any, List, Optional
from pydantic import BaseModel, HttpUrl

from app.models.manga_chapter_local import MangaChapterStatus


class MangaSeriesLocalRead(BaseModel):
    """本地漫画系列读取 Schema"""
    id: int
    source_id: int
    remote_series_id: str
    title: str
    alt_titles: Optional[List[str]] = None
    cover_url: Optional[str] = None  # 后端拼好的可访问 URL
    summary: Optional[str] = None
    authors: Optional[List[str]] = None
    tags: Optional[List[str]] = None
    status: Optional[str] = None
    language: Optional[str] = None
    remote_meta: Optional[dict[str, Any]] = None
    is_favorite: bool
    is_hidden: bool
    total_chapters: Optional[int] = None
    downloaded_chapters: Optional[int] = None
    last_sync_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class MangaChapterLocalRead(BaseModel):
    """本地漫画章节读取 Schema"""
    id: int
    series_id: int
    remote_chapter_id: str
    title: str
    number: Optional[float] = None
    volume: Optional[int] = None
    published_at: Optional[datetime] = None
    status: MangaChapterStatus
    page_count: Optional[int] = None
    last_error: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class LocalMangaPageRead(BaseModel):
    """本地漫画页面读取 Schema"""
    index: int
    image_url: str  # 相对静态路径，前端可直接使用

File: frontend/src/types/mangaLocal.ts
/**
 * 本地漫画库类型定义
 */

// import type { MangaSourceType } from './mangaSource'

export interface MangaSeriesLocal {
  id: number
  source_id: number
  remote_series_id: string
  title: string
  alt_titles?: string[]
  cover_url?: string  // 后端拼好的可访问 URL
  summary?: string
  authors?: string[]
  tags?: string[]
  status?: string
  language?: string
  is_favorite: boolean
  is_hidden: boolean
  total_chapters?: number
  downloaded_chapters?: number
  last_sync_at?: string
  created_at: string
  updated_at: string
}

export type MangaChapterStatus = 'PENDING' | 'DOWNLOADING' | 'READY' | 'FAILED'

export interface MangaChapterLocal {
  id: number
  series_id: number
  remote_chapter_id: string
  title: string
  number?: number
  volume?: number
  published_at?: string
  status: MangaChapterStatus
  page_count?: number
  last_error?: string | null
  created_at: string
  updated_at: string
}

export interface LocalMangaPage {
  index: number
  image_url: string
}

export interface MangaReadingProgress {
  id: number
  series_id: number
  chapter_id?: number | null
  last_page_index: number
  total_pages?: number | null
  is_finished: boolean
  last_read_at: string
  created_at: string
  updated_at: string
}

export interface MangaReadingHistoryItem {
  series_id: number
  series_title: string
  series_cover_url?: string
  source_name?: string
  last_chapter_id?: number
  last_chapter_title?: string
  last_page_index?: number
  total_pages?: number
  is_finished: boolean
  last_read_at: string
}

